internal/mage/engine.go

```diff
@@ -66,13 +66,15 @@ func (t Engine) Publish(ctx context.Context) error {
 	arches := []string{"amd64", "arm64"}
 	oses := []string{"linux", "darwin"}
 
-	imageRef, err := c.Container().Publish(ctx, engineImageRef, dagger.ContainerPublishOpts{
+	imageRef, err := c.Container().Publish(ctx, EngineImageRef, dagger.ContainerPublishOpts{
 		PlatformVariants: util.DevEngineContainer(c, arches, oses),
 	})
 	if err != nil {
 		return err
 	}
-	fmt.Println("Image published:", imageRef)
+
+	time.Sleep(3 * time.Second) // allow buildkit logs to flush, to minimize potential confusion
```

Comment:
Is it possible to check when logs are flushed? If yes, I would check periodically and fail on timeout. Sleeps like this one get worse the longer they are kept around.

Reply:
I'll double check quickly, but I think the answer is no, being able to check that will require a lot more changes to our logging related code (which should happen, just will take time0